src/inference/src/threading/ie_istreams_executor.cpp

```diff
@@ -46,51 +52,62 @@ int IStreamsExecutor::Config::GetDefaultNumStreams() {
-            config._threads_per_stream_big = config._threads_per_stream_small / 2;
-            config._big_core_streams = num_big_cores / config._threads_per_stream_big;
+            threads_per_stream_small = 0;
+        } else if (num_small_cores < threads_per_stream_small) {
+            small_core_streams = 1;
+            threads_per_stream_small = num_small_cores;
+            threads_per_stream_big = threads_per_stream_small;
+            // Balance the computation of physical core and logical core, the number of threads on the physical core and
+            // logical core should be equal
+            big_core_streams = num_big_cores_phy / threads_per_stream_big * 2;
         } else {
-            config._small_core_streams = num_small_cores / config._threads_per_stream_small;
+            small_core_streams = num_small_cores / threads_per_stream_small;
         }
     } else if (stream_mode == AGGRESSIVE) {
-        config._big_core_streams = num_big_cores;
-        config._small_core_streams = num_small_cores / 2;
-        config._threads_per_stream_big = num_big_cores / config._big_core_streams;
-        config._threads_per_stream_small = num_small_cores == 0 ? 0 : num_small_cores / config._small_core_streams;
+        big_core_streams = num_big_cores;
+        small_core_streams = num_small_cores;
+        threads_per_stream_big = num_big_cores / big_core_streams;
+        threads_per_stream_small = num_small_cores == 0 ? 0 : num_small_cores / small_core_streams;
     } else if (stream_mode == LESSAGGRESSIVE) {
-        config._big_core_streams = num_big_cores / 2;
-        config._small_core_streams = num_small_cores / 4;
-        config._threads_per_stream_big = num_big_cores / config._big_core_streams;
-        config._threads_per_stream_small = num_small_cores == 0 ? 0 : num_small_cores / config._small_core_streams;
+        big_core_streams = num_big_cores / 2;
+        small_core_streams = num_small_cores / 2;
+        threads_per_stream_big = num_big_cores / big_core_streams;
+        threads_per_stream_small = num_small_cores == 0 ? 0 : num_small_cores / small_core_streams;
     } else {
         IE_THROW() << "Wrong stream mode to get num of streams: " << stream_mode;
     }
-    config._small_core_offset = num_small_cores == 0 ? 0 : num_big_cores * 2;
-    return config._big_core_streams + config._small_core_streams;
+    config[CONFIG_KEY_INTERNAL(BIG_CORE_STREAMS)] = std::to_string(big_core_streams);
+    config[CONFIG_KEY_INTERNAL(SMALL_CORE_STREAMS)] = std::to_string(small_core_streams);
+    config[CONFIG_KEY_INTERNAL(THREADS_PER_STREAM_BIG)] = std::to_string(threads_per_stream_big);
+    config[CONFIG_KEY_INTERNAL(THREADS_PER_STREAM_SMALL)] = std::to_string(threads_per_stream_small);
+    // This is default setting for specific CPU which Pcore is in front and Ecore is in the back.
+    config[CONFIG_KEY_INTERNAL(SMALL_CORE_OFFSET)] = std::to_string(num_small_cores == 0 ? 0 : num_big_cores);
```

Comment:
num_big_cores is set as "num_big_cores_phy * 2". This might be not true for some specific HW or if hyper-threading is switched off manually.

Reply:
Ok, I added the judgment of whether there is a logical core.